src/librustc_traits/chalk/lowering.rs

```diff
@@ -78,10 +78,12 @@ impl<'tcx> LowerInto<'tcx, chalk_ir::InEnvironment<chalk_ir::Goal<RustInterner<'
     ) -> chalk_ir::InEnvironment<chalk_ir::Goal<RustInterner<'tcx>>> {
         let clauses = self.environment.into_iter().filter_map(|clause| match clause {
             ChalkEnvironmentClause::Predicate(predicate) => {
-                match predicate.kind() {
-                    ty::PredicateKind::Trait(predicate, _) => {
+                // FIXME(chalk): forall
+                match predicate.bound_atom(interner.tcx).skip_binder() {
```

Comment:
how is this different from `skip_binders()`?

Reply:
skip_binders doesn't shift unbound variables outwards